pitch.md

```diff
@@ -28,16 +31,28 @@ _How might the user interact with the language? What does programming look
 like? Why is this the right way to interact with the problem domain? Be careful
 to distinguish between, e.g,. a graphical interface and a linguistic interface._ 
 
+I am imagining a linguistic interface to start off but I also can picture, as my project develops, a GUI that maps directly to the linguistic interface (i dont think it would be too tough and would increase likelihood that it might be adopted by a broader range of teachers). I am imagining a syntax where at the top of the program the teacher specifies certain things such as the weight of different aspects of the constraint matcher (there would be defaults), students she would like to see work together or not work together, etc. This would be followed by a block of code for each student in which the relevant input data would be put in a formatted form something like:
+Name:
+Interests:
+Leadership interests:
+Commitment to A-level product:
+Desired Group Members:
+Undesired Group Members:
+
+In the GUI I imagine a bunch of drop downs or text boxes, dunno yet.
+
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of semantic (i.e., non-syntax) errors might occur, and how 
 might they be communicated to the user?_
+When a program runs it would be parsed into my constraint solver and then the logic would be run on it. Semantic erros might occur if people ask for contradictory constraints or if people ask for constraints that dont make sense. Those types of semantic errors are pretty relevant to most constraint solvers but not as much to my project because the _types_ of constraints allowed would be limited and since we are dealing with one type of object (students) then it would be pretty hard to get a constraint that didnt make sense. Both would still be possible though. I think I would communicate them to the user through command line errors.
 
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
+It should be _really, really easy_ to get groups of students based on their interests and preferences and I think it should be __impossible__ to do anything else. I could imagine that someone might be able to misuse it to solve some other type of constraint problem but I think it would be very difficult. I can't think of anything in the middle category of hard but possible.
```

Comment:
Fair enough, I don't think there's any reason to try to specifically design it so it couldn't be misused in that way.